File: kotlinjs/old/Builder.kt


abstract class Builder() : Unit() {
    override var occ = true
    override var traverse = false
    override var minCost = 50
    override var gasCost = 20
    override var nrgCost = 0
    override var commands = mutableListOf<String>("Build a Factory (200, 100, 0)", "Abort")
    override var health = 20
    override var minSal = 0
    override var gasSal = 0
    override var nrgSal = 0
}

File: dotnet/src/Model/Neutral/Units/Builder.cs
using System;
using System.Collections.Generic;

namespace PlanetCS1 {
    public abstract class Builder : Unit {
        public Builder() {
            this.occ = true;
            this.traverse = true;
            this.minCost = 50;
            this.gasCost = 20;
            this.nrgCost = 0;
            this.commands = new List<String>(){"Build a Factory (200, 100, 0)", "Abort"};
            this.health = 20;
            this.minSal = 0;
            this.gasSal = 0;
            this.nrgSal = 0;
        }
    }
}